apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/JavaResolver.kt

```diff
@@ -91,22 +107,22 @@ internal class JavaResolver(
       return if (generatePrimitiveTypes && type.ofType is IrScalarType) {
         resolveIrScalarType(type.ofType, asPrimitiveType = true)
       } else {
-        resolveIrType(type.ofType).let { if (wrapNullableFieldsInOptional) unwrapFromOptional(it) else it }
-      }
+        resolveIrType(type.ofType).let { if (wrapNullableFieldsInOptional) unwrapFromOptional(it) else it.withoutAnnotations() }
+      }.addNonNullableAnnotation()
     }
 
     return when (type) {
       is IrNonNullType -> error("") // make the compiler happy, this case is handled as a fast path
       is IrOptionalType -> resolveIrType(type.ofType).boxIfPrimitiveType().wrapInOptional()
-      is IrListType -> ParameterizedTypeName.get(JavaClassNames.List, resolveIrType(type.ofType).boxIfPrimitiveType())
+      is IrListType -> ParameterizedTypeName.get(JavaClassNames.List, resolveIrType(type.ofType).withoutAnnotations().boxIfPrimitiveType())
```

Comment:
I guess this means no `List<@NonNull Integer>`? This is fine for now but newest versions of `org.jetbrains:annotation` should support `TYPE_USE` so it'd be nice to have it at some point. Make a follow up issue?

Reply:
TIL! Didn't know this was allowed. Good idea! -> #4417